src/zarr/api/asynchronous.py

```diff
@@ -493,8 +493,33 @@ async def save_group(
     await asyncio.gather(*aws)
 
 
-async def tree(*args: Any, **kwargs: Any) -> None:
-    raise NotImplementedError
+async def tree(grp: AsyncGroup, expand: bool | None = None, level: int | None = None) -> Any:
```

Comment:
It's probably worth decorating this instead of putting the warning inside, like here:

https://github.com/zarr-developers/zarr-python/blob/0cca6b061e1f98594b206c69ecfc9f1c8972b4ea/src/zarr/core/group.py#L1085-L1088

I think this way IDEs and type checkers will pick up the deprecation if you try and use the method.